xcode/Subconscious/Shared/Parsers/Subtext.swift

```diff
@@ -662,19 +705,11 @@ extension Subtext {
             .prefix(2) // Take first two blocks
         
         let output = validBlocks.joined(separator: "\n")
+            .truncate(maxLength: Self.maxExcerptSize)
```

Comment:
The differences between this and `Subtext.truncate` are a source of inconsistencies.

I think we should factor the string truncation logic out of `Subtext.truncate`, and update `String.truncate` to use it, then use `String.truncate` in both places for consistent string truncation behavior.

Reply:
I've refactored the API surface, see https://github.com/subconsciousnetwork/subconscious/pull/936#discussion_r1368053902